---
 xpinstall/packager/Packager.pm |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: mozilla/xpinstall/packager/Packager.pm
===================================================================
--- mozilla.orig/xpinstall/packager/Packager.pm
+++ mozilla/xpinstall/packager/Packager.pm
@@ -142,17 +142,17 @@
     (-d "$srcdir/$line") && do {
       ($debug >= 10) && print "directory copy.\n";
       do_copydir ("$srcdir/$line");
       next LINE;
     };
 
     # if we hit this, it's either a file in the package file that is
     # not in the src directory, or it is not a valid entry.
-    warn "Warning: package error or possible missing or unnecessary file: $line ($package, $lineno).\n";
+    print "Warning: package error or possible missing or unnecessary file: $line ($package, $lineno).\n";
 
   } # LINE
 
   close (MANIFEST);
   chdir ($saved_cwd);
 
 }
 
