---
 directory/c-sdk/component_versions.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: seamonkey-2.0-2.0~a1~cvs20080123t1257+nobinonly/directory/c-sdk/component_versions.mk
===================================================================
--- seamonkey-2.0-2.0~a1~cvs20080123t1257+nobinonly.orig/directory/c-sdk/component_versions.mk
+++ seamonkey-2.0-2.0~a1~cvs20080123t1257+nobinonly/directory/c-sdk/component_versions.mk
@@ -41,17 +41,17 @@
 # NSPR - Netscape Portable Runtime
 NSPR_LIBVERSION		= 4
 NSPR_RELEASE_TAG	= v4.4.1
 NSPR_RELEASE_TREE	= $(COMPONENTS_DIR)
 
 # NSS - Network Security Services
 NSSVERS			= 3
 NSS_RELEASE_TAG		= NSS_3_9_3_RTM
-NSS_DYNAMIC_SOFTOKN	= 1
+NSS_DYNAMIC_SOFTOKN	= 0
 NSS_RELEASE_TREE	= $(COMPONENTS_DIR)
 
 # SVRCORE - Client/server utility library
 SVRCORE_RELEASE_TAG	= SVRCORE_4_0_RTM
 SVRCORE_RELEASE_TREE	= $(COMPONENTS_DIR)
 
 # LDAP library
 LDAPVERS		= 60
