diff -up thunderbird-3.0/mail/installer/Makefile.in.shared-error thunderbird-3.0/mail/installer/Makefile.in
--- thunderbird-3.0/mail/installer/Makefile.in.shared-error	2009-07-13 14:56:36.000000000 +0200
+++ thunderbird-3.0/mail/installer/Makefile.in	2009-07-14 12:41:16.000000000 +0200
@@ -94,9 +94,11 @@ endif
 # mozconfig instead.
 ifndef MAIL_PKG_SHARED
 ifndef BUILD_STATIC_LIBS
+ifeq (BUILD_STATIC_LIBS, 1)
 $(error you need an "--enable-static" build to package a build)
 endif
 endif
+endif
 
 include $(MOZILLA_SRCDIR)/toolkit/mozapps/installer/packager.mk
 
