--- mailnews/extensions/enigmail/package/install.rdf-orig	2010-02-14 20:31:15.203302835 -0600
+++ mailnews/extensions/enigmail/package/install.rdf	2010-02-14 20:31:35.005310657 -0600
@@ -15,7 +15,7 @@
       <Description>
         <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
         <em:minVersion>3.0</em:minVersion>
-        <em:maxVersion>3.0.*</em:maxVersion>
+        <em:maxVersion>3.1b1</em:maxVersion>
       </Description>
     </em:targetApplication>
     <em:targetApplication>
@@ -23,7 +23,7 @@
       <Description>
         <em:id>{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}</em:id>
         <em:minVersion>2.0.1</em:minVersion>
-        <em:maxVersion>2.0.*</em:maxVersion>
+        <em:maxVersion>2.1a1</em:maxVersion>
       </Description>
     </em:targetApplication>
     <em:targetApplication>
