--- xpcom/reflect/xptcall/src/md/unix/Makefile.in.orig	2007-11-24 20:28:19.000000000 +0000
+++ xpcom/reflect/xptcall/src/md/unix/Makefile.in	2007-11-24 20:28:34.000000000 +0000
@@ -278,7 +278,7 @@
 #
 # Linux/PPC
 #
-ifeq ($(OS_ARCH)$(OS_TEST),Linuxppc)
+ifeq ($(OS_ARCH)$(OS_TEST),Linuxpowerpc)
 CPPSRCS		:= xptcinvoke_ppc_linux.cpp xptcstubs_ppc_linux.cpp
 ASFILES		:= xptcinvoke_asm_ppc_linux.s xptcstubs_asm_ppc_linux.s
 AS		:= $(CC) -c -x assembler-with-cpp