Index: browser/locales/jar.mn
===================================================================
RCS file: /cvsroot/mozilla/browser/locales/jar.mn,v
retrieving revision 1.73
diff -p -u -8 -r1.73 jar.mn
--- browser/locales/jar.mn	29 Oct 2007 18:29:15 -0000	1.73
+++ browser/locales/jar.mn	7 Jan 2008 11:51:25 -0000
@@ -80,17 +80,20 @@
     locale/browser/help/using_firebird.xhtml          (%chrome/help/using_firebird.xhtml)
     locale/browser/help/customization.xhtml           (%chrome/help/customization.xhtml)
     locale/browser/help/firefox_welcome.xhtml         (%chrome/help/firefox_welcome.xhtml)
     locale/browser/help/download_manager.xhtml        (%chrome/help/download_manager.xhtml)
     locale/browser/help/platformStrings.dtd           (%chrome/help/platformStrings.dtd)
     locale/browser/help/tabbed_browsing.xhtml         (%chrome/help/tabbed_browsing.xhtml)
     locale/browser/help/glossary.xhtml                (%chrome/help/glossary.xhtml)
 # the following files are browser-specific overrides
-*   locale/@AB_CD@/global/netError.dtd                (%chrome/overrides/netError.dtd)
-*   locale/@AB_CD@/global/appstrings.properties       (%chrome/overrides/appstrings.properties)
-*   locale/@AB_CD@/mozapps/downloads/settingsChange.dtd  (%chrome/overrides/settingsChange.dtd)
+*   locale/browser/netError.dtd                (%chrome/overrides/netError.dtd)
+*   locale/browser/appstrings.properties       (%chrome/overrides/appstrings.properties)
+*   locale/browser/downloads/settingsChange.dtd  (%chrome/overrides/settingsChange.dtd)
+% override chrome://global/locale/netErrorApp.dtd chrome://browser/locale/netError.dtd
+% override chrome://global/locale/appstrings.properties chrome://browser/locale/appstrings.properties
+% override chrome://mozapps/locale/downloads/settingsChange.dtd chrome://browser/locale/downloads/settingsChange.dtd
 #ifdef MOZ_USE_GENERIC_BRANDING
 % locale branding @AB_CD@ %locale/branding/
     locale/branding/brand.dtd         (%chrome/branding/brand.dtd)
     locale/branding/brand.properties  (%chrome/branding/brand.properties)
 #endif
 #includesubst @LOCALE_SRCDIR@/extra-jar.mn
