--- mozilla/nsprpub/lib/libc/src/Makefile.in.orig	2005-06-01 14:28:26.000000000 +0000
+++ mozilla/nsprpub/lib/libc/src/Makefile.in	2006-01-15 02:50:39.000000000 +0000
@@ -112,6 +112,10 @@
 MKSHLIB += -R '$$ORIGIN'
 endif
 
+ifeq ($(OS_ARCH), Linux)
+DSO_LDOPTS +=-Wl,-R,'$$ORIGIN'
+endif
+
 ifeq ($(OS_ARCH),OS2)
 MAPFILE = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).def
 GARBAGE += $(MAPFILE)
--- mozilla/nsprpub/lib/ds/Makefile.in.orig	2005-06-01 14:28:25.000000000 +0000
+++ mozilla/nsprpub/lib/ds/Makefile.in	2006-01-15 02:52:30.000000000 +0000
@@ -102,6 +102,10 @@
 MKSHLIB += -R '$$ORIGIN'
 endif
 
+ifeq ($(OS_ARCH), Linux)
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
 ifeq ($(OS_ARCH),OS2)
 MAPFILE = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).def
 GARBAGE += $(MAPFILE)
--- mozilla/nsprpub/pr/src/Makefile.in.orig	2005-06-01 14:28:27.000000000 +0000
+++ mozilla/nsprpub/pr/src/Makefile.in	2006-01-15 03:29:36.000000000 +0000
@@ -168,6 +168,7 @@
 else
 OS_LIBS		= -ldl
 endif
+DSO_LDOPTS +=-Wl,-R,'$$ORIGIN'
 endif
 
 ifeq ($(OS_ARCH),HP-UX)
