--- toolkit/xre/nsAppRunner.cpp	2007-05-23 08:19:44.000000000 +0200
+++ toolkit/xre/nsAppRunner.cpp	2007-05-24 11:42:43.000000000 +0200
@@ -2435,16 +2435,17 @@ XRE_main(int argc, char* argv[], const n
     const char* desktopStartupIDEnv = PR_GetEnv("DESKTOP_STARTUP_ID");
     nsCAutoString desktopStartupID;
     if (desktopStartupIDEnv) {
       desktopStartupID.Assign(desktopStartupIDEnv);
     }
 #endif
 
 #if defined(MOZ_WIDGET_GTK2)
+    g_thread_init(NULL);
     // setup for private colormap.  Ideally we'd like to do this
     // in nsAppShell::Create, but we need to get in before gtk
     // has been initialized to make sure everything is running
     // consistently.
     if (CheckArg("install"))
       gdk_rgb_set_install(TRUE);
 
     // Initialize GTK here for splash
--- toolkit/library/Makefile.in	2007-05-03 08:25:37.000000000 +0200
+++ toolkit/library/Makefile.in	2007-05-26 01:18:45.000000000 +0200
@@ -187,17 +187,17 @@ EXTRA_DSO_LDOPTS += \
 	-framework QuickTime \
 	-framework IOKit \
 	-lcrypto \
 	$(TK_LIBS) \
 	$(NULL)
 endif
 
 ifneq (,$(filter gtk2 qt,$(MOZ_WIDGET_TOOLKIT)))
-EXTRA_DSO_LDOPTS += $(XLDFLAGS) $(XLIBS) $(MOZ_XFT_LIBS) $(MOZ_GTK2_LIBS) $(XT_LIBS)
+EXTRA_DSO_LDOPTS += $(XLDFLAGS) $(XLIBS) $(MOZ_XFT_LIBS) $(MOZ_GTK2_LIBS) $(XT_LIBS) -lgthread-2.0
 endif
 
 ifdef MOZ_ENABLE_XFT
 EXTRA_DSO_LDOPTS += $(FT2_LIBS)
 endif
 
 ifdef MOZ_ENABLE_STARTUP_NOTIFICATION
 EXTRA_DSO_LDOPTS += $(MOZ_STARTUP_NOTIFICATION_LIBS)
