--- security/manager/ssl/src/nsNSSComponent.cpp.orig	2007-05-27 21:04:27.000000000 +0200
+++ security/manager/ssl/src/nsNSSComponent.cpp	2007-05-27 21:17:56.000000000 +0200
@@ -130,6 +130,8 @@
 
 #define NS_CRYPTO_HASH_BUFFER_SIZE 4096
 
+#define GENTOO_NSS_LIBDIR  NS_LITERAL_CSTRING ("nss")
+
 static NS_DEFINE_CID(kNSSComponentCID, NS_NSSCOMPONENT_CID);
 int nsNSSComponent::mInstanceCount = 0;
 
@@ -708,11 +710,11 @@
     return;
 
   const char *possible_ckbi_locations[] = {
-    NS_XPCOM_CURRENT_PROCESS_DIR,
-    NS_GRE_DIR,
-    0 // This special value means: 
+    0,// This special value means: 
       //   search for ckbi in the directories on the shared
       //   library/DLL search path
+    NS_XPCOM_CURRENT_PROCESS_DIR,
+    NS_GRE_DIR
   };
 
   for (size_t il = 0; il < sizeof(possible_ckbi_locations)/sizeof(const char*); ++il) {
@@ -721,7 +723,16 @@
 
     if (!possible_ckbi_locations[il])
     {
-      fullLibraryPath = PR_GetLibraryName(nsnull, "nssckbi");
+    
+      directoryService->Get( NS_UNIX_LIB_DIR,
+                             NS_GET_IID(nsILocalFile), 
+                             getter_AddRefs(mozFile));
+
+      if (!mozFile) {
+        continue;
+      }
+
+      mozFile->AppendNative(GENTOO_NSS_LIBDIR);
     }
     else
     {
@@ -733,11 +744,12 @@
         continue;
       }
 
-      nsCAutoString processDir;
-      mozFile->GetNativePath(processDir);
-      fullLibraryPath = PR_GetLibraryName(processDir.get(), "nssckbi");
     }
 
+    nsCAutoString processDir;
+    mozFile->GetNativePath(processDir);
+    fullLibraryPath = PR_GetLibraryName(processDir.get(), "nssckbi");
+
     if (!fullLibraryPath) {
       continue;
     }
--- config/config.mk.orig	2007-05-27 23:32:35.000000000 +0200
+++ config/config.mk	2007-05-27 23:34:30.000000000 +0200
@@ -778,6 +778,7 @@
 
 DEFINES		+= -DOSTYPE=\"$(OS_CONFIG)\"
 DEFINES		+= -DOSARCH=$(OS_ARCH)
+DEFINES		+= -DGENTOO_LIB_DIR=\"$(libdir)\"
 
 # For profiling
 ifdef ENABLE_EAZEL_PROFILER
--- xpcom/io/SpecialSystemDirectory.cpp.orig	2007-05-27 23:36:29.000000000 +0200
+++ xpcom/io/SpecialSystemDirectory.cpp	2007-05-27 23:37:48.000000000 +0200
@@ -560,7 +560,7 @@
                                          PR_TRUE, 
                                          aFile);
         case Unix_LibDirectory:
-            return NS_NewNativeLocalFile(nsDependentCString("/usr/local/lib/netscape/"), 
+            return NS_NewNativeLocalFile(nsDependentCString(GENTOO_LIB_DIR), 
                                          PR_TRUE, 
                                          aFile);
 
