--- browser/app/Makefile.in.orig	2007-08-02 22:15:43.000000000 +0200
+++ browser/app/Makefile.in	2007-08-02 22:18:09.000000000 +0200
@@ -74,6 +74,16 @@
 
 ifdef LIBXUL_SDK
 include $(topsrcdir)/config/rules.mk
+libs::
+	$(INSTALL) $(DIST)/branding/default.xpm $(DIST)/bin/chrome/icons/default
+
+install::
+	$(SYSINSTALL) $(IFLAGS1) $(DIST)/branding/default.xpm $(DESTDIR)$(mozappdir)/chrome/icons/default
+export::
+ifndef MOZ_BRANDING_DIRECTORY
+	$(NSINSTALL) -D $(DIST)/branding
+	cp $(srcdir)/default.xpm   $(DIST)/branding/default.xpm	
+endif
 else
 # Build a binary bootstrapping with XRE_main
 
