--- config/autoconf.mk.in.orig	2008-07-08 15:12:31.000000000 +0000
+++ config/autoconf.mk.in	2008-07-08 15:13:10.000000000 +0000
@@ -58,14 +58,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/mozilla-thunderbird
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
 idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/mozilla-thunderbird
+sdkdir		= $(libdir)/mozilla-thunderbird
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
