diff --git a/xulrunner/app/nsRegisterGREUnix.cpp b/xulrunner/app/nsRegisterGREUnix.cpp
--- a/xulrunner/app/nsRegisterGREUnix.cpp
+++ b/xulrunner/app/nsRegisterGREUnix.cpp
@@ -41,6 +41,7 @@
 #include "nsXPCOM.h"
 #include "nsIFile.h"
 #include "nsILocalFile.h"
+#include "mozilla/FileUtils.h"
 
 #include "nsAppRunner.h" // for MAXPATHLEN
 #include "nsStringAPI.h"
@@ -53,13 +54,14 @@
 
 #include <unistd.h>
 #include <sys/stat.h>
-#include <mozilla/FileUtils.h>
 
 // If we can't register <buildid>.conf, we try to create a unique filename
 // by looping through <buildid>_<int>.conf, but if something is seriously wrong
 // we stop at 1000
 #define UNIQ_LOOP_LIMIT 1000
 
+using namespace mozilla;
+
 static const char kRegFileGlobal[] = "global.reginfo";
 static const char kRegFileUser[] = "user.reginfo";
 

