--- configure.in	(revision 6)
+++ configure.in	(local)
@@ -3835,7 +3835,21 @@
 
 fi # SKIP_LIBRARY_CHECKS
 
+dnl system MySpell Support
 dnl ========================================================
+MOZ_ARG_ENABLE_BOOL(system-myspell,
+[  --enable-system-myspell   Use system myspell (located with pkgconfig)],
+    SYSTEM_MYSPELL=1 )
+
+if test -n "$SYSTEM_MYSPELL"; then
+    PKG_CHECK_MODULES(MOZ_MYSPELL, hunspell)
+fi
+
+AC_SUBST(SYSTEM_MYSPELL)
+AC_SUBST(MOZ_MYSPELL_LIBS)
+AC_SUBST(MOZ_MYSPELL_CFLAGS)
+
+dnl ========================================================
 dnl Java SDK support
 dnl ========================================================
 JAVA_INCLUDE_PATH=
--- config/autoconf.mk.in	(revision 6)
+++ config/autoconf.mk.in	(local)
@@ -190,6 +190,10 @@
 NECKO_SMALL_BUFFERS = @NECKO_SMALL_BUFFERS@
 NECKO_COOKIES = @NECKO_COOKIES@
 
+MOZ_NATIVE_MYSPELL = @SYSTEM_MYSPELL@
+MOZ_MYSPELL_LIBS = @MOZ_MYSPELL_LIBS@
+MOZ_MYSPELL_CFLAGS = @MOZ_MYSPELL_CFLAGS@
+
 MOZ_NATIVE_ZLIB	= @SYSTEM_ZLIB@
 MOZ_NATIVE_JPEG	= @SYSTEM_JPEG@
 MOZ_NATIVE_PNG	= @SYSTEM_PNG@
--- extensions/spellcheck/src/Makefile.in.orig	2007-05-19 06:37:15.000000000 +0200
+++ extensions/spellcheck/src/Makefile.in	2007-05-19 06:47:02.000000000 +0200
@@ -62,6 +62,7 @@
 		  txtsvc \
 		  uconv \
 		  unicharutil \
+		  xulapp \
 		  $(NULL)
 
 CPPSRCS		= \
@@ -73,14 +74,28 @@
 		mozSpellI18NManager.cpp		\
 		mozInlineSpellChecker.cpp	\
 		mozInlineSpellWordUtil.cpp      \
-		$(NULL)
+		$(NULL)				\
+
+ifdef MOZ_NATIVE_MYSPELL
+CPPSRCS         += ../myspell/src/mozMySpell.cpp
+ifdef MOZ_XUL_APP
+CPPSRCS         += ../myspell/src/mozMySpellDirProvider.cpp
+endif
+endif
 
 ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 SHARED_LIBRARY_LIBS += ../osxspell/src/$(LIB_PREFIX)osxspell_s.$(LIB_SUFFIX)
 LOCAL_INCLUDES += -I$(srcdir)/../osxspell/src
 else
+ifndef MOZ_NATIVE_MYSPELL
 SHARED_LIBRARY_LIBS += ../myspell/src/$(LIB_PREFIX)myspell_s.$(LIB_SUFFIX)
 LOCAL_INCLUDES += -I$(srcdir)/../myspell/src
+else
+LOCAL_INCLUDES += $(MOZ_MYSPELL_CFLAGS) \
+		-I$(srcdir)/../myspell/src \
+		-DMOZ_NATIVE_MYSPELL \
+		$(NULL)
+endif
 endif
 
 EXTRA_DSO_LDOPTS = \
@@ -89,6 +101,10 @@
 		$(MOZ_UNICHARUTIL_LIBS) \
 		$(NULL)
 
+ifdef MOZ_NATIVE_MYSPELL
+EXTRA_DSO_LDOPTS += $(MOZ_MYSPELL_LIBS)
+endif
+
 include $(topsrcdir)/config/rules.mk
 
 LOCAL_INCLUDES += \
--- extensions/spellcheck/myspell/src/mozMySpell.h.orig	2007-05-19 06:48:00.000000000 +0200
+++ extensions/spellcheck/myspell/src/mozMySpell.h	2007-05-19 06:54:02.000000000 +0200
@@ -56,7 +56,12 @@
 #ifndef mozMySpell_h__
 #define mozMySpell_h__
 
+#ifdef MOZ_NATIVE_MYSPELL
+#include <hunspell.hxx>
+#else
 #include "myspell.hxx"
+#endif
+
 #include "mozISpellCheckingEngine.h"
 #include "mozIPersonalDictionary.h"
 #include "nsString.h"
@@ -104,7 +109,11 @@
   nsString  mDictionary;
   nsString  mLanguage;
 
+#ifdef MOZ_NATIVE_MYSPELL
+  Hunspell  *mMySpell;
+#else
   MySpell  *mMySpell;
+#endif
 };
 
 #endif
--- extensions/spellcheck/Makefile.in.orig	2007-05-19 07:52:47.000000000 +0200
+++ extensions/spellcheck/Makefile.in	2007-05-19 07:53:54.000000000 +0200
@@ -42,13 +42,20 @@
 include $(DEPTH)/config/autoconf.mk
 
 MODULE		= spellchecker
+
+ifndef MOZ_NATIVE_MYSPELL
 DIRS		= idl locales
+else
+DIRS		= idl
+endif
 
 ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 DIRS		+= osxspell
 else
+ifndef MOZ_NATIVE_MYSPELL
 DIRS		+= myspell
 endif
+endif
 
 DIRS += src
 
--- extensions/spellcheck/myspell/src/mozMySpell.cpp.orig	2007-05-20 05:06:02.000000000 +0200
+++ extensions/spellcheck/myspell/src/mozMySpell.cpp	2007-05-20 05:05:14.000000000 +0200
@@ -65,7 +65,10 @@
 #include "nsIDirectoryEnumerator.h"
 #include "nsDirectoryServiceUtils.h"
 #include "nsDirectoryServiceDefs.h"
+#ifdef MOZ_NATIVE_MYSPELL
+#include "nsILocalFile.h"
 #include "mozISpellI18NManager.h"
+#endif
 #include "nsICharsetConverterManager.h"
 #include "nsUnicharUtilCIID.h"
 #include "nsUnicharUtils.h"
@@ -150,8 +153,14 @@
 
   mDictionary = aDictionary;
 
+#ifdef MOZ_NATIVE_MYSPELL
+  mMySpell = new Hunspell(affFileName.get(),
+                         dictFileName.get());
+#else
   mMySpell = new MySpell(affFileName.get(),
                          dictFileName.get());
+#endif
+
   if (!mMySpell)
     return NS_ERROR_OUT_OF_MEMORY;
 
@@ -301,8 +310,14 @@
     do_GetService(NS_DIRECTORY_SERVICE_CONTRACTID);
   if (!dirSvc)
     return;
-
   nsCOMPtr<nsIFile> dictDir;
+#ifdef MOZ_NATIVE_MYSPELL
+   nsCOMPtr<nsILocalFile>  localFile;
+   rv = NS_NewNativeLocalFile(nsDependentCString("/usr/share/myspell"),PR_TRUE, getter_AddRefs(localFile));
+   if (localFile && NS_SUCCEEDED(rv))
+     localFile->QueryInterface(NS_GET_IID(nsIFile), getter_AddRefs(dictDir));
+   else {
+#endif
   rv = dirSvc->Get(DICTIONARY_SEARCH_DIRECTORY,
                    NS_GET_IID(nsIFile), getter_AddRefs(dictDir));
   if (NS_FAILED(rv)) {
@@ -314,6 +329,9 @@
 
     dictDir->AppendNative(NS_LITERAL_CSTRING("dictionaries"));
   }
+#ifdef MOZ_NATIVE_MYSPELL
+  }
+#endif
 
   LoadDictionariesFromDir(dictDir);
 
--- toolkit/library/Makefile.in.orig	2007-06-19 22:14:49.000000000 +0200
+++ toolkit/library/Makefile.in	2007-06-19 22:16:01.000000000 +0200
@@ -223,3 +223,8 @@
 dlldeps-javaxpcom.cpp: $(topsrcdir)/extensions/java/xpcom/src/dlldeps-javaxpcom.cpp
 	$(INSTALL) $^ .
 endif
+
+ifdef MOZ_NATIVE_MYSPELL
+EXTRA_DSO_LDOPTS += $(MOZ_MYSPELL_LIBS)
+endif
+
--- config/system-headers.orig	2007-07-23 02:14:54.000000000 +0200
+++ config/system-headers	2007-07-23 02:23:29.000000000 +0200
@@ -5,6 +5,8 @@
 AEPackObject.h
 AERegistry.h
 AEUtils.h
+affentry.hxx
+affixmgr.hxx
 afxcmn.h
 afxcoll.h
 afxcview.h
@@ -48,7 +50,9 @@
 atlwin.cpp
 ATSTypes.h
 ATSUnicode.h
+atypes.hxx
 Balloons.h
+baseaffix.hxx
 base/pblock.h
 base/PCR_Base.h
 base/session.h
@@ -139,6 +143,7 @@
 cstdio
 cstdlib
 cstring
+csutil.hxx
 ctype.h
 curses.h
 cxxabi.h
@@ -152,6 +157,7 @@
 descrip.h
 Devices.h
 Dialogs.h
+dictmgr.hxx
 direct.h
 dirent.h
 DiskInit.h
@@ -258,9 +264,13 @@
 gtk/gtkwidget.h
 gtk/gtkwindow.h
 gtk/gtkversion.h
+hashmgr.hxx
 HIToolbox/HIToolbox.h
 hlink.h
 htiface.h
+htypes.hxx
+hunspell.h
+hunspell.hxx
 ia64/sys/inline.h
 Icons.h
 iconv.h
@@ -303,6 +313,7 @@
 keyt.h
 LAction.h
 langinfo.h
+langnum.hxx
 LApplication.h
 LArray.h
 LArrayIterator.h
@@ -649,6 +660,7 @@
 strstrea.h
 structs.h
 stsdef.h
+suggestmgr.hxx
 SupportDefs.h
 support/String.h
 support/SupportDefs.h
