--- configure.in.orig	2007-08-06 23:12:31.000000000 +0200
+++ configure.in	2007-08-06 23:16:20.000000000 +0200
@@ -4026,6 +4026,21 @@
 
 fi # SKIP_LIBRARY_CHECKS
 
+dnl system Hunspell Support
+dnl ========================================================
+MOZ_ARG_ENABLE_BOOL(system-hunspell,
+[  --enable-system-hunspell   Use system hunspell (located with pkgconfig)],
+    SYSTEM_HUNSPELL=1 )
+
+if test -n "$SYSTEM_HUNSPELL"; then
+    PKG_CHECK_MODULES(MOZ_HUNSPELL, hunspell)
+fi
+
+AC_SUBST(SYSTEM_HUNSPELL)
+AC_SUBST(MOZ_HUNSPELL_LIBS)
+AC_SUBST(MOZ_HUNSPELL_CFLAGS)
+
+
 dnl check whether to enable glitz
 dnl ========================================================
 MOZ_ARG_ENABLE_BOOL(glitz,
--- config/autoconf.mk.in.orig	2007-08-06 23:16:37.000000000 +0200
+++ config/autoconf.mk.in	2007-08-06 23:18:31.000000000 +0200
@@ -201,6 +201,10 @@
 NECKO_COOKIES = @NECKO_COOKIES@
 MOZ_AUTH_EXTENSION = @MOZ_AUTH_EXTENSION@
 
+MOZ_NATIVE_HUNSPELL = @SYSTEM_HUNSPELL@
+MOZ_HUNSPELL_LIBS = @MOZ_HUNSPELL_LIBS@
+MOZ_HUNSPELL_CFLAGS = @MOZ_HUNSPELL_CFLAGS@
+
 MOZ_NATIVE_ZLIB	= @SYSTEM_ZLIB@
 MOZ_NATIVE_JPEG	= @SYSTEM_JPEG@
 MOZ_NATIVE_PNG	= @SYSTEM_PNG@
--- config/system-headers.orig	2007-08-06 23:37:32.000000000 +0200
+++ config/system-headers	2007-08-06 23:37:36.000000000 +0200
@@ -5,6 +5,8 @@
 AEPackObject.h
 AERegistry.h
 AEUtils.h
+affentry.hxx
+affixmgr.hxx
 afxcmn.h
 afxcoll.h
 afxcview.h
@@ -48,7 +50,9 @@
 atlwin.cpp
 ATSTypes.h
 ATSUnicode.h
+atypes.hxx
 Balloons.h
+baseaffix.hxx
 base/pblock.h
 base/PCR_Base.h
 base/session.h
@@ -139,6 +143,7 @@
 cstdio
 cstdlib
 cstring
+csutil.hxx
 ctype.h
 curses.h
 cxxabi.h
@@ -152,6 +157,7 @@
 descrip.h
 Devices.h
 Dialogs.h
+dictmgr.hxx
 direct.h
 dirent.h
 DiskInit.h
@@ -258,9 +264,13 @@
 gtk/gtkwidget.h
 gtk/gtkwindow.h
 gtk/gtkversion.h
+hashmgr.hxx
 HIToolbox/HIToolbox.h
 hlink.h
 htiface.h
+htypes.hxx
+hunspell.h
+hunspell.hxx
 ia64/sys/inline.h
 Icons.h
 iconv.h
@@ -303,6 +313,7 @@
 keyt.h
 LAction.h
 langinfo.h
+langnum.hxx
 LApplication.h
 LArray.h
 LArrayIterator.h
@@ -649,6 +660,7 @@
 strstrea.h
 structs.h
 stsdef.h
+suggestmgr.hxx
 SupportDefs.h
 support/String.h
 support/SupportDefs.h
--- extensions/spellcheck/hunspell/src/mozHunspell.cpp.orig	2007-08-06 23:26:05.000000000 +0200
+++ extensions/spellcheck/hunspell/src/mozHunspell.cpp	2007-08-06 23:31:30.000000000 +0200
@@ -63,6 +63,9 @@
 #include "nsISimpleEnumerator.h"
 #include "nsIDirectoryEnumerator.h"
 #include "nsIFile.h"
+#ifdef MOZ_NATIVE_HUNSPELL
+#include "nsILocalFile.h"
+#endif
 #include "nsDirectoryServiceUtils.h"
 #include "nsDirectoryServiceDefs.h"
 #include "mozISpellI18NManager.h"
@@ -307,6 +310,13 @@
     return;
 
   nsCOMPtr<nsIFile> dictDir;
+#ifdef MOZ_NATIVE_HUNSPELL
+  nsCOMPtr<nsILocalFile>  localFile;
+  rv = NS_NewNativeLocalFile(nsDependentCString("/usr/share/myspell"),PR_TRUE, getter_AddRefs(localFile));
+  if (localFile && NS_SUCCEEDED(rv))
+    localFile->QueryInterface(NS_GET_IID(nsIFile), getter_AddRefs(dictDir));
+  else {
+#endif
   rv = dirSvc->Get(DICTIONARY_SEARCH_DIRECTORY,
                    NS_GET_IID(nsIFile), getter_AddRefs(dictDir));
   if (NS_FAILED(rv)) {
@@ -318,6 +328,9 @@
 
     dictDir->AppendNative(NS_LITERAL_CSTRING("dictionaries"));
   }
+#ifdef MOZ_NATIVE_HUNSPELL
+  }
+#endif
 
   LoadDictionariesFromDir(dictDir);
 
--- extensions/spellcheck/Makefile.in.orig	2007-08-06 23:24:44.000000000 +0200
+++ extensions/spellcheck/Makefile.in	2007-08-06 23:43:16.000000000 +0200
@@ -42,13 +42,19 @@
 include $(DEPTH)/config/autoconf.mk
 
 MODULE		= spellchecker
+ifndef MOZ_NATIVE_HUNSPELL
 DIRS		= idl locales
+else
+DIRS		= idl
+endif
 
 ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 DIRS		+= osxspell
 else
+ifndef MOZ_NATIVE_HUNSPELL
 DIRS		+= hunspell
 endif
+endif
 
 DIRS += src
 
--- toolkit/library/Makefile.in.orig	2007-08-06 23:32:05.000000000 +0200
+++ toolkit/library/Makefile.in	2007-08-06 23:33:13.000000000 +0200
@@ -238,6 +238,10 @@
 	$(INSTALL) $^ .
 endif
 
+ifdef MOZ_NATIVE_HUNSPELL
+EXTRA_DSO_LDOPTS += $(MOZ_HUNSPELL_LIBS)
+endif
+
 ifneq (,$(filter layout-debug,$(MOZ_EXTENSIONS)))
 DEFINES += -DMOZ_ENABLE_EXTENSION_LAYOUT_DEBUG
 endif
--- extensions/spellcheck/src/Makefile.in.orig	2007-08-07 22:01:18.000000000 +0200
+++ extensions/spellcheck/src/Makefile.in	2007-08-07 22:01:54.000000000 +0200
@@ -62,9 +62,16 @@
 		  txtsvc \
 		  uconv \
 		  unicharutil \
+		  xulapp \
 		  $(NULL)
 
-CPPSRCS		= \
+ifdef MOZ_NATIVE_HUNSPELL
+CPPSRCS         += mozHunspell.cpp
+ifdef MOZ_XUL_APP
+CPPSRCS         += mozHunspellDirProvider.cpp
+endif
+endif
+CPPSRCS		+= \
 		mozSpellCheckerFactory.cpp	\
 		mozSpellChecker.cpp		\
 		mozPersonalDictionary.cpp	\
@@ -79,8 +86,14 @@
 SHARED_LIBRARY_LIBS += ../osxspell/src/$(LIB_PREFIX)osxspell_s.$(LIB_SUFFIX)
 LOCAL_INCLUDES += -I$(srcdir)/../osxspell/src
 else
+ifndef MOZ_NATIVE_HUNSPELL
 SHARED_LIBRARY_LIBS += ../hunspell/src/$(LIB_PREFIX)hunspell_s.$(LIB_SUFFIX)
 LOCAL_INCLUDES += -I$(srcdir)/../hunspell/src
+else
+LOCAL_INCLUDES += $(MOZ_HUNSPELL_CFLAGS) \
+		-DMOZ_NATIVE_HUNSPELL \
+		$(NULL)
+endif
 endif
 
 EXTRA_DSO_LDOPTS = \
@@ -89,8 +102,11 @@
 		$(MOZ_UNICHARUTIL_LIBS) \
 		$(NULL)
 
-include $(topsrcdir)/config/rules.mk
+ifdef MOZ_NATIVE_HUNSPELL
+EXTRA_DSO_LDOPTS += $(MOZ_HUNSPELL_LIBS)
+endif
 
+include $(topsrcdir)/config/rules.mk
 LOCAL_INCLUDES += \
 	-I$(topsrcdir)/xpcom/io \
 	$(NULL)
@@ -101,3 +117,17 @@
 	-framework Foundation \
 	$(NULL)
 endif
+ifdef MOZ_NATIVE_HUNSPELL
+export:: $(srcdir)/../hunspell/src/mozHunspell.cpp
+	$(INSTALL) $(srcdir)/../hunspell/src/mozHunspell.* .
+GARBAGE += mozHunspell.$(OBJ_SUFFIX) mozHunspell.cpp
+clean::
+	rm -f mozHunspell.*
+ifdef MOZ_XUL_APP
+export:: $(srcdir)/../hunspell/src/mozHunspellDirProvider.cpp
+	$(INSTALL) $(srcdir)/../hunspell/src/mozHunspellDirProvider.* .
+GARBAGE += mozHunspellDirProvider.$(OBJ_SUFFIX) mozHunspellDirProvider.cpp
+clean::
+	rm -f mozHunspellDirProvider.*
+endif
+endif
