--- extensions/wallet/build/Makefile.in.orig	2007-06-10 22:58:58.000000000 +0200
+++ extensions/wallet/build/Makefile.in	2007-06-10 23:00:25.000000000 +0200
@@ -48,7 +48,7 @@
 SHORT_LIBNAME	= wlltvwrs
 IS_COMPONENT	= 1
 MODULE_NAME	= nsWalletViewerModule
-MOZILLA_INTERNAL_API = 1
+LIBXUL_LIBRARY = 1
 
 PACKAGE_FILE = walletviewer.pkg
 
@@ -73,6 +73,8 @@
 		$(MOZ_COMPONENT_LIBS) \
 		$(NULL)
 
+
+#		$(XPCOM_STANDALONE_GLUE_LDOPTS)
 ifeq ($(OS_ARCH),WINNT)
 OS_LIBS		+= $(call EXPAND_LIBNAME,shell32)
 endif
--- extensions/wallet/src/Makefile.in.orig	2007-06-10 22:59:17.000000000 +0200
+++ extensions/wallet/src/Makefile.in	2007-06-10 22:59:44.000000000 +0200
@@ -48,7 +48,7 @@
 IS_COMPONENT	= 1
 MODULE_NAME	= nsWalletModule
 GRE_MODULE	= 1
-MOZILLA_INTERNAL_API = 1
+LIBXUL_LIBRARY = 1
 
 PACKAGE_FILE = wallet.pkg
 
@@ -94,6 +94,7 @@
 		$(MOZ_COMPONENT_LIBS) \
 		$(NULL)
 
+#		$(XPCOM_STANDALONE_GLUE_LDOPTS)
 include $(topsrcdir)/config/rules.mk
 
 libs::
