--- embedding/browser/gtk/src/gtkmozembed_common.cpp.orig	2007-08-04 13:27:48.000000000 +0200
+++ embedding/browser/gtk/src/gtkmozembed_common.cpp	2007-08-04 13:28:05.000000000 +0200
@@ -78,7 +78,7 @@
 #include "nsIDOMNavigator.h"
 #include "nsIDOMPluginArray.h"
 #include "nsIDOMPlugin.h"
-#include <plugin/nsIPluginHost.h>
+#include "nsIPluginHost.h"
 #include "nsIDOMMimeType.h"
 #include "nsIObserverService.h"
 
