Add proper support for xulrunner-1.9.2 system nss/nspr

--- xulrunner/installer/libxul-embedding.pc.in	2009-11-11 11:20:38.000000000 -0600
+++ xulrunner/installer/libxul-embedding.pc.in	2009-11-11 11:22:10.000000000 -0600
@@ -7,4 +7,4 @@
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue
-Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS%
+Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS% %FULL_NSPR_CFLAGS%
--- xulrunner/installer/Makefile.in	2009-11-11 11:20:38.000000000 -0600
+++ xulrunner/installer/Makefile.in	2009-11-11 11:26:56.000000000 -0600
@@ -94,19 +94,28 @@
 ifdef MOZ_NATIVE_NSPR
 NSPR_NAME=nspr
 NSPR_VERSION=$(shell $(NSPR_CONFIG) --version)
+FULL_NSPR_CFLAGS=$(shell $(NSPR_CONFIG) --cflags)
+FULL_NSPR_LIBS=$(shell $(NSPR_CONFIG) --libs)
 else
-pkg_config_files += mozilla-nspr.pc
 NSPR_NAME=mozilla-nspr
 FULL_NSPR_CFLAGS=-I\$${includedir}
 FULL_NSPR_LIBS=$(subst $(prefix),\$${sdkdir},$(shell $(DEPTH)/nsprpub/config/nspr-config --libs))
 NSPR_VERSION=$(shell $(DEPTH)/nsprpub/config/nspr-config --version)
 endif
+npkg_config_files += mozilla-nspr.pc
 
 $(warning FULL_NSPR_CFLAGS=$(FULL_NSPR_CFLAGS))
 
-ifndef MOZ_NATIVE_NSS
-pkg_config_files += mozilla-nss.pc
+ifdef MOZ_NATIVE_NSS
+NSS_VERSION=$(shell $(NSS_CONFIG) --version)
+FULL_NSS_CFLAGS=$(shell $(NSS_CONFIG) --cflags)
+FULL_NSS_LIBS=$(shell $(NSS_CONFIG) --libs)
+else
+NSS_VERSION=$(shell $(DEPTH)/nsprpub/config/nss-config --version)
+FULL_NSS_CFLAGS=-I\$${includedir}/stable
+FULL_NSS_LIBS=$(subst $(prefix),\$${sdkdir},$(shell $(DEPTH)/nsprpub/config/nss-config --libs))
 endif
+pkg_config_files += mozilla-nss.pc
 
 ifdef MOZ_OJI
 OJIDEF=-DOJI
@@ -127,7 +136,10 @@
 	-e "s|%FULL_NSPR_LIBS%|$(FULL_NSPR_LIBS)|" \
 	-e "s|%FULL_NSPR_CFLAGS%|$(FULL_NSPR_CFLAGS)|" \
 	-e "s|%NSPR_NAME%|$(NSPR_NAME)|" \
-	-e "s|%NSPR_VERSION%|$(NSPR_VERSION)|" > $@
+	-e "s|%NSPR_VERSION%|$(NSPR_VERSION)|" \
+	-e "s|%FULL_NSS_LIBS%|$(FULL_NSS_LIBS)|" \
+	-e "s|%FULL_NSS_CFLAGS%|$(FULL_NSS_CFLAGS)|" \
+	-e "s|%NSS_VERSION%|$(NSS_VERSION)|" > $@
 	chmod 644 $@
 
 install:: $(pkg_config_files)
--- xulrunner/installer/mozilla-nss.pc.in	2009-11-11 11:20:38.000000000 -0600
+++ xulrunner/installer/mozilla-nss.pc.in	2009-11-11 11:22:13.000000000 -0600
@@ -4,7 +4,7 @@
 
 Name: NSS
 Description: Mozilla Network Security Services
-Version: %MOZILLA_VERSION%
+Version: %NSS_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Libs: -L${sdkdir}/lib -lsmime3 -lssl3 -lnss3 -lnssutil3
-Cflags: -I${includedir}
+Libs: %FULL_NSS_LIBS%
+Cflags: %FULL_NSS_CFLAGS%
