Add proper support for xulrunner-1.9.2 system nspr

--- xulrunner/installer/libxul-embedding.pc.in	2009-11-11 11:20:38.000000000 -0600
+++ xulrunner/installer/libxul-embedding.pc.in	2009-11-11 11:22:10.000000000 -0600
@@ -6,5 +6,6 @@
 Name: libxul-embedding
 Description: Static library for version-independent embedding of the Mozilla runtime
 Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
 Libs: -L${sdkdir}/lib -lxpcomglue
 Cflags: -DXPCOM_GLUE -I${includedir} %WCHAR_CFLAGS%
\
