--- rdf/util/src/Makefile.in.orig	2007-08-01 01:26:41.000000000 +0200
+++ rdf/util/src/Makefile.in	2007-08-01 01:26:45.000000000 +0200
@@ -63,6 +63,10 @@
 
 CPPSRCS		= $(RDF_UTIL_SRC_LCPPSRCS)
 
+SDK_LIBRARY     =                        \
+		$(LIB_PREFIX)rdfutil_external_s.$(LIB_SUFFIX) \
+		$(NULL)
+
 # we don't want the shared lib, but we want to force the creation of a static lib.
 FORCE_STATIC_LIB = 1
 
--- intl/unicharutil/util/internal/Makefile.in.orig	2007-08-01 01:23:44.000000000 +0200
+++ intl/unicharutil/util/internal/Makefile.in	2007-08-01 01:24:44.000000000 +0200
@@ -66,6 +66,10 @@
 
 CPPSRCS	= $(INTL_UNICHARUTIL_UTIL_LCPPSRCS)
 
+SDK_LIBRARY     =                        \
+		$(LIB_PREFIX)unicharutil_s.$(LIB_SUFFIX) \
+		$(NULL)
+
 FORCE_STATIC_LIB = 1
 FORCE_USE_PIC = 1
 
--- xpcom/typelib/xpt/src/Makefile.in.orig	2007-08-01 01:20:34.000000000 +0200
+++ xpcom/typelib/xpt/src/Makefile.in	2007-08-01 01:21:53.000000000 +0200
@@ -54,6 +54,10 @@
 CSRCS		= xpt_arena.c xpt_struct.c xpt_xdr.c
 HOST_CSRCS	= $(CSRCS)
 
+SDK_LIBRARY     =                        \
+		$(LIB_PREFIX)xpt.$(LIB_SUFFIX) \
+		$(NULL)
+
 # we don't want the shared lib, but we want to force the creation of a static lib.
 FORCE_STATIC_LIB = 1
 
