--- xulrunner/installer/Makefile.in.orig	2007-08-05 12:22:08.000000000 +0200
+++ xulrunner/installer/Makefile.in	2007-08-05 12:19:19.000000000 +0200
@@ -74,15 +74,20 @@
 	libxul.pc \
 	libxul-embedding.pc \
 	mozilla-js.pc \
+	xulrunner-js.pc \
 	mozilla-plugin.pc \
+	xulrunner-plugin.pc \
+	xulrunner-xpcom.pc \
 	$(NULL)
 
 ifdef MOZ_NATIVE_NSPR
 NSPR_NAME=nspr
 NSPR_VERSION=$(shell $(NSPR_CONFIG) --version)
 pkg_config_files += mozilla-nspr.pc
+pkg_config_files += xulrunner-nspr.pc
 else
 pkg_config_files += mozilla-nspr.pc
+pkg_config_files += xulrunner-nspr.pc
 NSPR_NAME=mozilla-nspr
 FULL_NSPR_CFLAGS=-I\$${includedir}/stable
 FULL_NSPR_LIBS=$(subst $(prefix),\$${sdkdir},$(shell $(DEPTH)/nsprpub/config/nspr-config --libs))
@@ -98,6 +98,7 @@
 
 ifdef MOZ_NATIVE_NSS
 pkg_config_files += mozilla-nss.pc
+pkg_config_files += xulrunner-nss.pc
 NSS_VERSION=$(shell $(NSS_CONFIG) --version)
 NSS_NAME=nss
 endif
@@ -103,6 +108,7 @@
 	-e "s|%includedir%|$(includedir)|" \
 	-e "s|%idldir%|$(idldir)|" \
 	-e "s|%sdkdir%|$(sdkdir)|" \
+	-e "s|%libdir%|$(sdkdir)|" \
 	-e "s|%MOZ_APP_NAME%|$(MOZ_APP_NAME)|" \
 	-e "s|%MOZILLA_VERSION%|$(MOZ_APP_VERSION)|" \
 	-e "s|%FULL_NSPR_LIBS%|$(FULL_NSPR_LIBS)|" \
--- /dev/null	2007-08-04 01:52:23.948566729 +0200
+++ xulrunner/installer/xulrunner-js.pc.in	2007-08-05 12:03:08.000000000 +0200
@@ -0,0 +1,11 @@
+prefix=%prefix%
+sdkdir=%sdkdir%
+libdir=%sdkdir%
+includedir=%includedir%/unstable
+
+Name: JavaScript
+Description: The Mozilla JavaScript Library - fake   pkg-config file
+Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
+Libs: -L${sdkdir} -lmozjs
+Cflags: -I${includedir} -DXP_UNIX -DJS_THREADSAFE
--- /dev/null	2007-08-04 01:52:23.948566729 +0200
+++ xulrunner/installer/xulrunner-nspr.pc.in	2007-08-05 12:01:00.000000000 +0200
@@ -0,0 +1,6 @@
+Name: NSPR
+Description: The Netscape Portable Runtime
+Version: %NSPR_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
+
+
--- /dev/null	2007-08-04 01:52:23.948566729 +0200
+++ xulrunner/installer/xulrunner-nss.pc.in	2007-08-05 12:01:09.000000000 +0200
@@ -0,0 +1,4 @@
+Name: NSS
+Description: Mozilla Network Security Services
+Version: %NSS_VERSION%
+Requires: %NSS_NAME% >= %NSS_VERSION%
--- /dev/null	2007-08-04 01:52:23.948566729 +0200
+++ xulrunner/installer/xulrunner-plugin.pc.in	2007-08-05 12:03:54.000000000 +0200
@@ -0,0 +1,10 @@
+prefix=%prefix%
+sdkdir=%sdkdir%
+libdir=%sdkdir%
+includedir=%includedir%/unstable
+
+Name: Mozilla Plug-In API
+Description: Mozilla Plug-In API
+Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
+Cflags: -I${includedir}
--- /dev/null	2007-08-04 01:52:23.948566729 +0200
+++ xulrunner/installer/xulrunner-xpcom.pc.in	2007-08-05 12:04:59.000000000 +0200
@@ -0,0 +1,16 @@
+prefix=%prefix%
+sdkdir=%sdkdir%
+libdir=%sdkdir%
+includedir=%includedir%/unstable
+idldir=%idldir%/unstable
+
+# Note: the default pkg-config includes will only list include directories
+# that contain stable (frozen or safe for linking) header files. To include
+# nonfrozen headers, run pkg-config with --define-variable=includetype=unstable
+
+Name: libxul
+Description: The Mozilla Runtime and Embedding Engine
+Version: %MOZILLA_VERSION%
+Requires: %NSPR_NAME% >= %NSPR_VERSION%
+Libs: -L${sdkdir} -lxpcomglue_s -lxul -lxpcom
+Cflags: -I${includedir} -fshort-wchar
