--- configure.in.old	2007-12-10 12:29:54 +0000
+++ configure.in	2007-12-10 12:38:19 +0000
@@ -2862,6 +2862,7 @@
     ;;
 *)
     AC_CHECK_LIB(m, atan)
+    AC_CHECK_HEADER(dlfcn.h, AC_DEFINE(HAVE_DLCNF_H))
     AC_CHECK_LIB(dl, dlopen,
     AC_CHECK_HEADER(dlfcn.h, 
         LIBS="-ldl $LIBS"
--- xpcom/base/nsStackWalk.cpp.old	2007-12-10 12:43:02 +0000
+++ xpcom/base/nsStackWalk.cpp	2007-12-10 12:43:51 +0000
@@ -1111,7 +1111,7 @@
 #define __USE_GNU
 #endif
 
-#if defined(HAVE_LIBDL) || defined(XP_MACOSX)
+#if defined(HAVE_DLCNF_H) || defined(XP_MACOSX)
 #include <dlfcn.h>
 #endif
 
