--- intl/uconv/native/nsNativeUConvService.cpp.old	2007-12-10 13:04:29 +0000
+++ intl/uconv/native/nsNativeUConvService.cpp	2007-12-10 13:04:58 +0000
@@ -229,7 +229,7 @@
 
     do {
         if ( iconv(mConverter,
-                   (char **)&aSrc,
+                   &aSrc,
                    &inLeft,
                    (char **)&aDest,
                    &outLeft) == (size_t) -1 ) {
@@ -375,7 +375,7 @@
 
     do {
         if ( iconv(mConverter,
-                   (char **)&aSrc,
+                   (const char **)&aSrc,
                    &inLeft,
                    (char **)&aDest,
                    &outLeft) == (size_t) -1 ) {
