Mozilla bug 368844
--- configure.in	2007-02-12 05:04:12.000000000 +0100
+++ configure.in.new	2007-02-21 15:42:01.000000000 +0100
@@ -7225,10 +7225,10 @@
    MOZ_TREE_CAIRO=,
    MOZ_TREE_CAIRO=1 )
 
-   if test "$MOZ_TREE_CAIRO"; then
-       # Check for headers defining standard int types.
-       AC_CHECK_HEADERS(stdint.h inttypes.h sys/int_types.h)
+   # Check for headers defining standard int types.
+   AC_CHECK_HEADERS(stdint.h inttypes.h sys/int_types.h)
 
+   if test "$MOZ_TREE_CAIRO"; then
        # For now we assume that we will have a uint64_t available through
        # one of the above headers or mozstdint.h.
        AC_DEFINE(HAVE_UINT64_T)
