# HG changeset patch
# Parent 94a51a3b64d40175c2d32419cbe467604f183f4d
Bug 629638 - Build elfhack injected code with a limited set of CFLAGS

diff --git a/build/unix/elfhack/Makefile.in b/build/unix/elfhack/Makefile.in
--- a/build/unix/elfhack/Makefile.in
+++ b/build/unix/elfhack/Makefile.in
@@ -102,16 +102,17 @@ endif
 inject:
 	$(NSINSTALL) -D $@
 
 inject/%.c: inject.c | inject
 	cp $< $@
 
 GARBAGE_DIRS += inject
 
-inject/%.$(OBJ_SUFFIX): DEFINES += -DBITS=$(if $(HAVE_64BIT_OS),64,32)
-inject/$(CPU)-noinit.$(OBJ_SUFFIX): DEFINES += -DNOINIT
-
 # need this to suppress errors due to /usr/include/linux/byteorder/swab.h
 # on mozilla buildbots
 OS_CXXFLAGS := $(filter-out -pedantic,$(OS_CXXFLAGS))
 
 include $(topsrcdir)/config/rules.mk
+
+inject/%.$(OBJ_SUFFIX): DEFINES += -DBITS=$(if $(HAVE_64BIT_OS),64,32)
+inject/%.$(OBJ_SUFFIX): CFLAGS := -O2 -fno-stack-protector $(filter -m% -I%,$(CFLAGS))
+inject/$(CPU)-noinit.$(OBJ_SUFFIX): DEFINES += -DNOINIT

