--- configure.in.orig	2010-10-08 21:41:12.000000000 +0200
+++ configure.in	2010-10-20 21:28:26.191633777 +0200
@@ -4858,7 +4858,7 @@
 dnl =========================================================
 MOZ_ARG_DISABLE_BOOL(mailnews,
 [  --disable-mailnews      Disable building of mail & news components],
-    MOZ_MAIL_NEWS=,
+    MOZ_MAIL_NEWS=0,
     MOZ_MAIL_NEWS=1 )
 AC_SUBST(MOZ_MAIL_NEWS)
 
@@ -4867,7 +4867,7 @@
 dnl =========================================================
 MOZ_ARG_DISABLE_BOOL(ldap,
 [  --disable-ldap          Disable LDAP support],
-    MOZ_LDAP_XPCOM=,
+    MOZ_LDAP_XPCOM=0,
     MOZ_LDAP_XPCOM=1)
 
 dnl ========================================================
