--- mozilla/modules/libpr0n/decoders/nsPNGDecoder.cpp
+++ mozilla/modules/libpr0n/decoders/nsPNGDecoder.cpp
@@ -386,7 +386,8 @@
   // First try to see if iCCP chunk is present
   if (png_get_valid(png_ptr, info_ptr, PNG_INFO_iCCP)) {
     png_uint_32 profileLen;
-    char *profileData, *profileName;
+    png_bytep profileData;
+    png_charp profileName;
     int compression;
 
     png_get_iCCP(png_ptr, info_ptr, &profileName, &compression,
