---
 suite/installer/unix/packages |   11 +++++++++++
 1 file changed, 11 insertions(+)

Index: mozilla/suite/installer/unix/packages
===================================================================
--- mozilla.orig/suite/installer/unix/packages
+++ mozilla/suite/installer/unix/packages
@@ -43,25 +43,27 @@
 [browser]
 bin/@MOZ_APP_NAME@-bin
 bin/@MOZ_APP_NAME@
 bin/application.ini
 bin/license.txt
 bin/mozilla-xremote-client
 bin/platform.ini
 bin/plugins/libnullplugin.so
+bin/plugins/libunixprintplugin.so
 bin/README
 bin/run-mozilla.sh
 
 ; shared libraries
 bin/libgfxpsshar.so
 bin/libgkgfx.so
 bin/libgtkxtbin.so
 bin/libjsj.so
 bin/libsqlite3.so
+bin/libjemalloc.so
 
 ; Components
 bin/components/libaccessibility.so
 bin/components/accessibility.xpt
 bin/components/alerts.xpt
 bin/components/libappcomps.so
 bin/components/appshell.xpt
 bin/components/libnsappshell.so
@@ -144,16 +146,17 @@
 bin/components/libjsd.so
 bin/components/jsdservice.xpt
 bin/components/layout_base.xpt
 bin/components/layout_printing.xpt
 bin/components/layout_xul.xpt
 bin/components/layout_xul_tree.xpt
 bin/components/locale.xpt
 bin/components/loginmgr.xpt
+bin/components/places.xpt
 bin/components/lwbrk.xpt
 bin/components/mimetype.xpt
 bin/components/libmork.so
 bin/components/mozbrwsr.xpt
 bin/components/libmozfind.so
 bin/components/mozfind.xpt
 bin/components/libnecko.so
 bin/components/necko.xpt
@@ -212,16 +215,17 @@
 bin/components/libuniversalchardet.so
 bin/components/update.xpt
 bin/components/uriloader.xpt
 bin/components/urlformatter.xpt
 bin/components/webBrowser_core.xpt
 bin/components/webbrowserpersist.xpt
 bin/components/libwebbrwsr.so
 bin/components/webshell_idls.xpt
+bin/components/websrvcs.xpt
 bin/components/widget.xpt
 bin/components/libwidget_gtk2.so
 bin/components/libwindowds.so
 bin/components/windowds.xpt
 bin/components/windowwatcher.xpt
 bin/components/libxpautocomplete.so
 bin/components/xpautocomplete.xpt
 bin/components/xpcom_base.xpt
@@ -272,22 +276,28 @@
 bin/components/nsTryToClose.js
 bin/components/nsUpdateService.js
 bin/components/nsURLFormatter.js
 bin/components/nsWebHandlerApp.js
 bin/components/pluginGlue.js
 bin/components/storage-Legacy.js
 bin/components/storage-mozStorage.js
 bin/components/txEXSLTRegExFunctions.js
+bin/components/nsInterfaceInfoToIDL.js
 
 ; Modules
 bin/modules/*
 bin/modules/gloda/*
 
 ; Safe Browsing
+bin/components/nsSafebrowsingApplication.js
+bin/components/safebrowsing.xpt
+bin/components/nsUrlClassifierListManager.js
+bin/components/nsUrlClassifierLib.js
+bin/components/url-classifier.xpt
 
 ; Kerberos NegotiateAuth
 bin/components/libauth.so
 
 ; GNOME hooks
 bin/components/libmozgnome.so
 
 ; Browser Chrome Files
@@ -350,16 +360,17 @@
 bin/defaults/autoconfig/*
 bin/defaults/pref/*
 bin/greprefs/*
 
 ; LDAP components need to be in the browser for AutoConfig
 bin/libldap60.so
 bin/libldif60.so
 bin/libprldap60.so
+bin/libssldap60.so
 bin/components/libmozldap.so
 bin/components/mozldap.xpt
 bin/components/nsLDAPProtocolHandler.js
 
 ; [Layout Engine Resources]
 ; Style Sheets, Graphics and other Resources used by the layout engine.
 bin/res/arrow.gif
 bin/res/arrowd.gif
