Index: ./toolkit/components/passwordmgr/src/nsLoginManagerPrompter.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/components/passwordmgr/src/nsLoginManagerPrompter.js,v
retrieving revision 1.30
diff -u -p -8 -r1.30 nsLoginManagerPrompter.js
--- a/toolkit/components/passwordmgr/src/nsLoginManagerPrompter.js	9 Mar 2008 02:25:40 -0000	1.30
+++ b/toolkit/components/passwordmgr/src/nsLoginManagerPrompter.js	9 Mar 2008 23:33:24 -0000
@@ -560,18 +560,25 @@ LoginManagerPrompter.prototype = {
               this._getLocalizedString("notifyBarRememberButtonText");
         var rememberButtonAccessKey =
               this._getLocalizedString("notifyBarRememberButtonAccessKey");
         var notNowButtonText =
               this._getLocalizedString("notifyBarNotNowButtonText");
         var notNowButtonAccessKey =
               this._getLocalizedString("notifyBarNotNowButtonAccessKey");
 
-        var brandShortName =
-              this._brandBundle.GetStringFromName("brandShortName");
+        // XXX asac: anything more reasonable than "" we can guess for
+        // embedders that don't provide branding?)
+        var brandShortName =  "";
+        try {
+                brandShortName =
+                      this._brandBundle.GetStringFromName("brandShortName");
+        } catch (e) {
+                this.log ("no brandShortName available for prompting - most likely incomplete embedding.");
+        }
         var notificationText  = this._getLocalizedString(
                                         "savePasswordText", [brandShortName]);
 
         // The callbacks in |buttons| have a closure to access the variables
         // in scope here; set one to |this._pwmgr| so we can get back to pwmgr
         // without a getService() call.
         var pwmgr = this._pwmgr;
 
