Index: nsprpub/pr/include/prerror.h
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/include/prerror.h,v
retrieving revision 3.11.4.2
diff -u -8 -r3.11.4.2 prerror.h
--- mozilla/nsprpub/pr/include/prerror.h	2 Apr 2007 09:08:49 -0000	3.11.4.2
+++ mozilla/nsprpub/pr/include/prerror.h	25 May 2007 19:30:50 -0000
@@ -44,31 +44,16 @@
 
 typedef PRInt32 PRErrorCode;
 
 #define PR_NSPR_ERROR_BASE -6000
 
 #include "prerr.h"
 
 /*
-** Compile-time assert. "condition" must be a constant expression.
-** The macro should be used only once per source line in places where
-** a "typedef" declaration is allowed.
-** For stringification of the line numbers where the macro is used we need some
-** macro indirection. IMPL is required to get macro-expansion of __LINE__ to
-** its integer value so that IMPL2 will stringify the number, not "__LINE__".
-*/
-#define PR_STATIC_ASSERT(condition)                                           \
-    PR_STATIC_ASSERT_IMPL(condition, __LINE__)
-#define PR_STATIC_ASSERT_IMPL(condition, line)                                \
-    PR_STATIC_ASSERT_IMPL2(condition, line)
-#define PR_STATIC_ASSERT_IMPL2(condition, line)                               \
-    typedef int pr_static_assert_line_##line[(condition) ? 1 : -1]
-
-/*
 ** Set error will preserve an error condition within a thread context.
 ** The values stored are the NSPR (platform independent) translation of
 ** the error. Also, if available, the platform specific oserror is stored.
 ** If there is no appropriate OS error number, a zero my be supplied.
 */
 NSPR_API(void) PR_SetError(PRErrorCode errorCode, PRInt32 oserr);
 
 /*
Index: nsprpub/pr/include/prlog.h
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/pr/include/prlog.h,v
retrieving revision 3.13.2.1
diff -u -8 -r3.13.2.1 prlog.h
--- mozilla/nsprpub/pr/include/prlog.h	28 Apr 2004 00:33:43 -0000	3.13.2.1
+++ mozilla/nsprpub/pr/include/prlog.h	25 May 2007 19:30:50 -0000
@@ -246,11 +246,19 @@
 
 #else
 
 #define PR_ASSERT(expr) ((void) 0)
 #define PR_NOT_REACHED(reasonStr)
 
 #endif /* defined(DEBUG) || defined(FORCE_PR_ASSERT) */
 
+/*
+** Compile-time assert. "condition" must be a constant expression.
+** The macro can be used only in places where an "extern" declaration is
+** allowed.
+*/
+#define PR_STATIC_ASSERT(condition) \
+    extern void pr_static_assert(int arg[(condition) ? 1 : -1])
+
 PR_END_EXTERN_C
 
 #endif /* prlog_h___ */
