--- a/build/autoconf/nss.m4	2015-11-20 15:37:26.951993242 -0500
+++ b/build/autoconf/nss.m4	2015-11-20 15:37:19.431992752 -0500
@@ -8,6 +8,8 @@
 AC_DEFUN([AM_PATH_NSS],
 [dnl
 
+PKG_CHECK_MODULES([NSS],[nss >= $1],[ifelse([$2], , :, [$2])],[
+
 AC_ARG_WITH(nss-prefix,
 	[  --with-nss-prefix=PFX   Prefix where NSS is installed],
 	nss_config_prefix="$withval",
@@ -84,8 +86,9 @@
 		ifelse([$3], , :, [$3])
 	fi
 
+	])
 
-	AC_SUBST(NSS_CFLAGS)
+	AC_SUBST_LIST(NSS_CFLAGS)
 	AC_SUBST_LIST(NSS_LIBS)
 
 ])
--- a/configure.in	2015-11-20 15:40:57.401762087 -0500
+++ b/configure.in	2015-11-20 15:50:10.080854755 -0500
@@ -8775,7 +8775,7 @@
 AC_SUBST(NSPR_CFLAGS)
 AC_SUBST(MOZ_NATIVE_NSPR)
 
-AC_SUBST(NSS_CFLAGS)
+AC_SUBST_LIST(NSS_CFLAGS)
 AC_SUBST(MOZ_NATIVE_NSS)
 AC_SUBST(NSS_DISABLE_DBM)
 
